### PR TITLE
[WIP]update vm docs

### PR DESCRIPTION
**Pull request by irisdingbj**
co-authored by @sdake 

THis is the draft doc update for VM. Please help double check. Many Thanks.

### CODE REVIEW

File: content/en/docs/examples/virtual-machines/multi-network/index.md [hunk 1]
```diff
@@ -75,22 +68,31 @@ The following example updates the `/etc/hosts` file with the {{< gloss >}}Istiod
    /etc/hosts is an easy to use example. It is also possible to use a real DNS and certificate for Istiod, this is beyond
    the scope of this document.
 
+2. Obtain `tcp-istiod` port for Istio ingress gateway and export it as `CAPORT`. Add it into `/var/lib/istio/envoy/sidecar.env`.  Revisit the [preparing the cluster](#preparing-the-kubernetes-cluster-for-vms) section to learn how to obtain the `tcp-istiod` port.
```
Comment (frankbu) [inline content/en/docs/examples/virtual-machines/multi-network/index.md:71]:
```suggestion
1. Obtain `tcp-istiod` port for Istio ingress gateway and export it as `CAPORT`. Add it into `/var/lib/istio/envoy/sidecar.env`.  Revisit the [preparing the cluster](#preparing-the-kubernetes-cluster-for-vms) section to learn how to obtain the `tcp-istiod` port.
```

Comment (frankbu) [reply content/en/docs/examples/virtual-machines/multi-network/index.md:71]: > Revisit the [preparing the cluster](#preparing-the-kubernetes-cluster-for-vms) section to learn how to obtain the `tcp-istiod` port.

I don't see any instructions about obtaining `tcp-istiod` port at that link.

Comment (irisdingbj) [reply content/en/docs/examples/virtual-machines/multi-network/index.md:71]: @frankbu the two :$CAPORT are referencing `tcp-istiod` port
   {{< text bash >}}
    $ echo "CA_ADDR=istiod.istio-system.svc:$CAPORT" >> cluster.env
    $ echo "ISTIO_PILOT_PORT=$CAPORT >> cluster.env
    {{< /text >}}

Comment (frankbu) [reply content/en/docs/examples/virtual-machines/multi-network/index.md]: @irisdingbj Yes, I see that, but there is no instructions for how to set CAPORT, even though it says "Revisit the preparing the cluster section ..." to learn how to.

File: content/en/docs/examples/virtual-machines/multi-network/index.md [hunk 2]
```diff
@@ -75,22 +67,31 @@ The following example updates the `/etc/hosts` file with the {{< gloss >}}Istiod
...
 
+1. Obtain `tcp-istiod` port for Istio ingress gateway and export it as `CAPORT`. Add it into `/var/lib/istio/envoy/sidecar.env`.  Revisit the [preparing the cluster](#preparing-the-kubernetes-cluster-for-vms) section to learn how to obtain the `tcp-istiod` port.
+
+    {{< text bash >}}
+    $ echo "CA_ADDR=istiod.istio-system.svc:$CAPORT" >> cluster.env
+    $ echo "ISTIO_PILOT_PORT=$CAPORT >> cluster.env
+    {{< /text >}}
+
+    {{< tip >}}
+    Ensure you have used @.name=="tcp-istiod" to extract the `tcp-istiod` port for Istio ingress gateway
```
Comment (frankbu) [inline content/en/docs/examples/virtual-machines/multi-network/index.md:78]: Unclear what this means. What is "@.name"? Ensure you used it in which instructions step?

Comment (irisdingbj) [reply content/en/docs/examples/virtual-machines/multi-network/index.md:78]: thanks for catching this, I fixed the reference link.

**Comment by frankbu** [comment]
@irisdingbj What is the status of this PR? Is it still WIP? If not, please remove the WIP and do-not-merge label.

Also, you still have not addressed my concern about there not being any clear instruction about how to set the CAPORT env variable https://github.com/istio/istio.io/pull/7107#discussion_r412462167. You point to similar instructions that set HTTP/HTTPS ports, but it is surely a leap for the reader to understand that you mean they should use a similar kubectl command to what is done there to set a different env variable for a different port of istiod. You need to explain what they should do, or the reader will be lost.

**Comment by sdake** [comment]
> @irisdingbj What is the status of this PR? Is it still WIP? If not, please remove the WIP and do-not-merge label.
> 
> Also, you still have not addressed my concern about there not being any clear instruction about how to set the CAPORT env variable [#7107 (comment)](https://github.com/istio/istio.io/pull/7107#discussion_r412462167). You point to similar instructions that set HTTP/HTTPS ports, but it is surely a leap for the reader to understand that you mean they should use a similar kubectl command to what is done there to set a different env variable for a different port of istiod. You need to explain what they should do, or the reader will be lost.

@frankbu Iris is on holiday until the 10th. I had agreed with her to take over this PR prior to her holiday, although I find myself in a time crunch as well. This work will land - but it may be very late in the review process.

I'll pull down the review and start work tomorrow to improve this documentation. ATM I am working to unblock the development toolchain.

Cheers
-steve

**Comment by dex80526** [comment]
@irisdingbj: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: content/en/docs/examples/virtual-machines/single-network/index.md [hunk 3]
```diff
@@ -138,7 +114,7 @@ following commands on a machine with cluster admin privileges:
 
     {{< text bash >}}
     $ go run istio.io/istio/security/tools/generate_cert \
-          -client -host spiffee://cluster.local/vm/vmname --out-priv key.pem --out-cert cert-chain.pem  -mode citadel
+          -client -server -host spiffe://cluster.local/ns/vm/sa/default  --out-priv key.pem --out-cert cert-chain.pem  -mode citadel
```
Comment (FrimIdan) [inline content/en/docs/examples/virtual-machines/single-network/index.md:117]: @irisdingbj this command will work only if istiod is running self-signed mode, right?

**Comment by sdake** [comment]
@irisdingbj and I have worked on https://github.com/istio/istio.io/pull/7248. @irisdingbj did most of the testing, and I did most of the writing. If you have an interest here as a contributor of any type, feel free to create an improvement PR.

For now, closing this PR.

**Comment by sdake** [comment]
@FrimIdan can you take a look at https://github.com/istio/istio.io/tree/master/content/en/docs/setup/install/virtual-machine. If you have any followup please contact the docs wg in #docs in Istio's slack. Alternately feel free to open improvement PRs.

Cheers
-steve

**Comment by FrimIdan** [comment]
@sdake Thanks, I will take a look.

**Pull request closed without merging**